frontend/packages/topology/src/components/export-app/export-app-context.ts

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { AlertVariant } from '@patternfly/react-core';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import {
+  useActiveNamespace,
+  USERSETTINGS_PREFIX,
+  useToast,
+  useUserSettings,
+} from '@console/shared/src';
+import { exportAppUserSettings } from './types';
+
+export const ExportAppContext = React.createContext({});
+
+export const ExportAppContextProvider = ExportAppContext.Provider;
+
+export const useExportAppFormToast = () => {
+  const toast = useToast();
+  const { t } = useTranslation();
+  const [namespace] = useActiveNamespace();
+  const [exportAppToast, setExportAppToast, exportAppToastLoaded] = useUserSettings<
+    exportAppUserSettings
+  >(`${USERSETTINGS_PREFIX}.exportApp`, {}, true);
+
+  const exportAppWatchResources = React.useMemo(() => {
+    if (!exportAppToastLoaded) return {};
+    const keys = Object.keys(exportAppToast);
+    let watchRes = {};
+    keys.forEach((k) => {
+      const { kind, name, namespace: resNamespace } = exportAppToast[k];
+      watchRes = {
+        ...watchRes,
+        [k]: {
+          kind,
+          name,
+          namespace: resNamespace,
+          namespaced: true,
+          isList: false,
+          optional: true,
+        },
+      };
+    });
+    return watchRes;
+  }, [exportAppToast, exportAppToastLoaded]);
+
+  const extraResources = useK8sWatchResources<{ [k: string]: K8sResourceKind }>(
+    exportAppWatchResources,
+  );
+
+  const callbackFunc = React.useCallback(
+    (routeUrl) => {
```

Comment:
pass exported namespace
```suggestion
    (routeUrl, exported_namespace) => {
```